src/components/ui/QRCode/QRCode.tsx

```diff
@@ -0,0 +1,69 @@
+import html2canvas from 'html2canvas';
+import { useRef, useState } from 'react';
+import QRCodeComponent from 'react-qr-code';
+import { AnimatePresence, motion } from 'framer-motion';
+
+const QRCode = ({ url }: { url: string }) => {
+  const [isHovered, setIsHovered] = useState(false);
+  const qrCodeRef = useRef<HTMLDivElement>(null);
+
+  const downloadAsImage = () => {
+    const element = qrCodeRef.current as HTMLDivElement;
+
+    html2canvas(element).then((canvas) => {
```

Comment:
We should think about giving the downloaded QR file a meaningful name. Something that tells you what it is. How much, etc

Reply:
Yes. I thought about it but it was getting really long. One suggestion would be nf-PR-{PaymentReqestID}. Adding amount in the name would be tricky with decimal.

Reply:
It would be worth ask Andres and Pablo what they think.

Reply:
Updated PR. Changed the file name format to QR-DD-MM-YYYY-HHMMSS.